### DISCUSSION TITLE
Lottie files

### DISCUSSION THREAD

**Discussion by shakamoushie** [zauberzeug/nicegui#262] [Ideas / Feature Requests]
Does NiceGUI have a provision to display lottie files? If yes, could you share sample code?

Thanks in advance..

**Comment by rodja** [comment]
Yes, NiceGUI allows to use any html/js:

```python
from nicegui import ui

ui.add_body_html('<script src="https://unpkg.com/@lottiefiles/lottie-player@latest/dist/lottie-player.js"></script>')

ui.html('''
<lottie-player src="https://assets1.lottiefiles.com/datafiles/HN7OcWNnoqje6iXIiZdWzKxvLIbfeCGTmvXmEm1h/data.json"  
background="transparent"  speed="1" style="width: 300px; height: 300px;" loop autoplay></lottie-player>
''')

ui.run()
```

It also would not be to hard to create a seperate `ui.lottie` element similar to [the map example where leaflet.js is wrapped](https://github.com/zauberzeug/nicegui/tree/main/examples/map). Would you like to try this?

**Comment by shakamoushie** [comment]
Hi @rodja,

Thank you for your prompt response. Your example code works very well with online lottie files. However, with offline files, I can't seem to get it to work. My code is below:

```py
from nicegui import ui
import json

vpth = "D:/Python/Tryouts/nicegui/"

def PlayLottie(vFile, vHeight=500, vWidth=700):
    try:
        with open(vpth + vFile, "r") as fl:
            LottieCode = json.load(fl)

        htmlstr = f"<lottie-player src='{LottieCode}', background='transparent' speed='1' style='width: {vWidth}px; height: {vHeight}px;' loop autoplay></lottie-player>"
        ui.html(htmlstr)

    except:
        ui.label(f"Lottie load error for {vFile}")

ui.add_body_html('<script src="https://unpkg.com/@lottiefiles/lottie-player@latest/dist/lottie-player.js"></script>')
PlayLottie("typing.json", 300, 300)

ui.run()
```

**Reply by falkoschindler** [reply]
You can't pass the Lottie _content_ to the `src` attribute. Instead you will need to serve the file, e.g. using [app.add_static_files](https://nicegui.io/reference#add_static_files), and set `src` to its path:

```py
from nicegui import app, ui

app.add_static_files('/lottie', 'lottie')
ui.add_body_html('<script src="https://unpkg.com/@lottiefiles/lottie-player@latest/dist/lottie-player.js"></script>')
ui.html('<lottie-player src="lottie/loading.json", background="transparent" speed="1" loop autoplay></lottie-player>')

ui.run()
```

**Reply by shakamoushie** [reply]
Thank you @falkoschindler, this works very well for local lottie files.

**Comment by falkoschindler** [comment]
We converted this Q&A into an Idea because it would be nice to have built-in support for Lottie files in NiceGUI.

**Comment by rodja** [comment]
We just added a lottie demo to [ui.icon](https://nicegui.io/documentation/icon#lottie_files) and [ui.image](https://nicegui.io/documentation/image#lottie_files).
